python/ray/serve/_private/router.py

```diff
@@ -311,21 +399,64 @@ async def select_from_candidate_replicas(
         # In that case, return `None` so a new one is selected.
         return self._replicas.get(chosen_replica_id, None)
 
-    def fulfill_next_pending_assignment(self, replica: ReplicaWrapper):
-        """Assign the replica to the next pending assignment in FIFO order.
+    def _get_pending_request_matching_metadata(
+        self,
+        replica: ReplicaWrapper,
+        request_metadata: Optional[RequestMetadata] = None,
+    ) -> Optional[PendingRequest]:
+        if request_metadata is None or not request_metadata.multiplexed_model_id:
+            return None
+
+        for pr in self._pending_requests_to_fulfill:
```

Comment:
This feels a bit inefficient, but I think we should leave it as-is for now. Each time we try to fulfill a pending multiplex request, we have to iterate through the queue of all pending requests.

In the future, we may want to optimize this using a skiplist or some other data structure.

Reply:
yes, though in the common case the request w/ the requested model_id should be towards the front of the list